python/tests/test_delete.py

```diff
@@ -0,0 +1,57 @@
+import pathlib
+
+import pyarrow as pa
+import pyarrow.compute as pc
+
+from deltalake.table import DeltaTable
+from deltalake.writer import write_deltalake
+
+
+def test_delete_no_predicates(existing_table: DeltaTable):
+    old_version = existing_table.version()
+
+    existing_table.delete()
+
+    last_action = existing_table.history(1)[0]
+    assert last_action["operation"] == "DELETE"
+    assert existing_table.version() == old_version + 1
+
+    dataset = existing_table.to_pyarrow_dataset()
+    assert dataset.count_rows() == 0
+    assert len(existing_table.files()) == 0
+
+
+def test_delete_a_partition(tmp_path: pathlib.Path, sample_data: pa.Table):
+    write_deltalake(tmp_path, sample_data, partition_by=["bool"])
+
+    dt = DeltaTable(tmp_path)
+    old_version = dt.version()
+
+    expr = pc.field("bool") != pc.scalar(True)
+    expected_table = sample_data.filter(expr)
```

Comment:
This doesn't work in older versions of PyArrow, which we test with. Perhaps instead it would be sufficient to do:

```suggestion
    mask = pc.equal(sample_data['bool'], False)
    expected_table = sample_data.filter(make)
```